manifests/install.pp

```diff
@@ -413,23 +441,47 @@
       Package["${python_pkg_prefix}-six"],
     ] + $redhat_7_deps,
   }
-  ->python::pip { 'cobald':
-    ensure       => present,
-    pkgname      => $pip_cobald,
-    virtualenv   => $::cobald::params::virtualenv,
-    url          => $cobald_url,
-    owner        => 'root',
-    timeout      => 1800,
-    require      => $piprequire,
+  ->if $cobald_repo_type == 'git' {
+    python::pip { 'cobald':
+      ensure       => present,
+      pkgname      => $pip_cobald,
+      virtualenv   => $::cobald::params::virtualenv,
+      url          => $cobald_url,
+      owner        => 'root',
+      timeout      => 1800,
+      require      => $piprequire,
+    }
+  } else {
+    python::pip { 'cobald':
+      ensure       => present,
+      pkgname      => $pip_cobald,
+      virtualenv   => $::cobald::params::virtualenv,
+      index        => $cobald_url,
+      owner        => 'root',
+      timeout      => 1800,
+      require      => $piprequire,
+    }
```

Comment:
I think this distinction of cases (`if $cobald_repo_type == 'git' ... else ...`) is not necessary or do I miss anything?

Reply:
I also don't see a reason for the distinction of cases. Dropping it will also remove the syntax `-> if` which is not so nice to read. As a general rule, the most readable practice seems to be to store the differences of cases in attributes defined before the resource or in a hash, and then apply that to the ressource (which is actually exactly what you do with `pip_cobald` and `cobald_url` :wink:  ).